docs/cardinal/commands.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: Commands
+---
+
+Commands are how players interact with your game world. They represent actions that players want to perform, such as moving, attacking, or chatting.
+
+To handle commands in your game, you must first define them and create systems to handle them. Commands can come from either game clients or other shards.
+
+## Defining Commands
+
+Commands are plain Go structs that embed `BaseCommand` and implement the `Command` interface. This interface requires a single `Name()` method that returns a **unique** string identifier.
+
+```go
+import "github.com/argus-labs/world-engine/pkg/cardinal"
+
+type AttackCommand struct {
+    cardinal.BaseCommand // Required
+    TargetID string `json:"target_id"`
+    Damage   int    `json:"damage"`
+}
+
+func (AttackCommand) Name() string {
+    return "attack"
+}
+```
+
+## Handling Commands
+
+Just like with components, you must declare the commands a system can access in its state struct. Add a `WithCommand[T]` field to your system state type, where `T` is your command type:
+
+```go
+import "github.com/argus-labs/world-engine/pkg/cardinal"
+
+type AttackSystemState struct {
+    cardinal.BaseSystemState
+    AttackCommands cardinal.WithCommand[AttackCommand]
+}
+```
+
+<Note>
+  A system cannot have multiple `WithCommand[T]` fields with the same command type. However, different systems can handle the same command type. This is useful for when you want a single command to trigger multiple game logic and/or side effects.
+</Note>
+
+### Iterating Over Commands
+
+Use `Iter` to loop through all commands received this tick. Each iteration yields a `CommandContext` that provides access to the command data and metadata.
+
+To access the command's payload, use `Payload`:
+
+```go
+func AttackSystem(state *AttackSystemState) error {
+    for cmd := range state.AttackCommands.Iter() {
+        attack := cmd.Payload()
+        // Use attack.TargetID, attack.Damage, etc.
+    }
+    return nil
+}
+```
+
+### Personas
+
+A persona is the unique identity associated with a player’s account. All commands include metadata containing the sender’s persona, which you can use for authorization checks like verifying entity ownership.
```

Comment:
If one reads the docs starting from Introduction and follow the Next buttons, one doesn’t get introduced to anything that leads to Persona, therefore Persona/account/authorization feel foreign, with no link to learn more. Perhaps a `<Tip>` component could help here.

Reply:
Makes sense. I wanted to add a section to explain Cardinal's architecture more, but I'm not sure where to put it in. The "the world" section at the top of the ECS page was originally supposed to be here in a "introduction" page, but I feel that going to ECS first is better to get readers more hands-on/involved (e.g. they can try on their own while reading) vs putting an introduction/architecture page first that is heavy on theory will be too "academic"(?). academic isn't the right word but I'm not sure how best to describe it. Going straight to ECS is more "quick-starty" than learning the high-level architecture as that can be learned later.